src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -350,13 +349,22 @@ def validate(self, node, options=None):
                     f"must be full ranges but found {n_full_ranges}.")
 
         # Make sure the result has as many full range as needed
-        if result.children:
-            for idx, child in enumerate(result.children):
-                if isinstance(child, Range) and not result.is_full_range(idx):
-                    raise TransformationError(
-                        f"To use {self.name} on matmul, each range on "
-                        f"the result variable '{result.name}' must be a full "
-                        f"range but found {result.debug_string()}")
+        if len(result.symbol.shape) in [1, 2] and not result.children:
```

Comment:
We now have this check three times (for 'result', 'matrix1' and 'matrix2'). Please could you refactor so we just have one utility routine that does this check - you can pass in the permitted number of ranges and the text for which the check is for.

Reply:
Addressed in 6fb1e5a1ffda53c4a956266e070b3a161dda58d0. The text was fairly simple to homogenise so I opted not to pass it through.